File: python/fast-pyhton/prime.cpp
#include <iostream>

bool is_prime(int n) {
  bool bl = true;
  for (int j=2; j<n; j++) {
    if (n%j == 0) {
      bl = false;
      break;
    }
  }
  return bl;
}

int main() {
  int n = 50000;
  int cnt = 0;
  for (int i=1; i<=n; i++) {
    if (is_prime(i)) cnt++;
  }
  std::cout << cnt << std::endl;
  return 0;
}

File: python/fast-pyhton/prime.py
import cprime


def main():
    n = 50000
    cnt = 0
    for i in range(n):
        if cprime.isPrime(i + 1):
            cnt += 1

    print(cnt)


def isPrime(n):
    bool_ = True
    for i in range(2, n):
        if n % i == 0:
            bool_ = False
            break
    return bool_


def isPrimeSmart(n):
    bool_ = True
    for i in range(2, int(n**1 / 2) + 1):
        if n % i == 0:
            bool_ = False
            break
    return bool_


if __name__ == "__main__":
    main()
